packages/nx/src/command-line/connect/connect-to-nx-cloud.ts

```diff
@@ -14,9 +14,9 @@ export async function connectToNxCloudIfExplicitlyAsked(opts: {
 }): Promise<void> {
   if (opts['cloud'] === true) {
     const nxJson = readNxJson();
-    const runners = Object.values(nxJson.tasksRunnerOptions);
+    const runners = Object.values(nxJson.tasksRunnerOptions ?? {});
     const onlyDefaultRunnerIsUsed =
-      runners.length === 1 && runners[0].runner === 'nx/tasks-runners/default';
+      runners.length <= 1 && runners[0].runner === 'nx/tasks-runners/default';
     if (!onlyDefaultRunnerIsUsed) return;
```

Comment:
This logic is wrong when `taskRunnerOptions` is not defined.

The `length` of the values will be 0.